File: src/main/java/com/felipe/booking/app/dto/BookingDTO.java
package com.felipe.booking.app.dto;

import com.felipe.booking.domain.model.Booking;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
public class BookingDTO {
    private LocalDateTime checkInDate;
    private LocalDateTime checkOutDate;

    public static BookingDTO of(Booking booking) {
        return new BookingDTO(booking.getCheckInDate(), booking.getCheckOutDate());
    }

    public Booking toDomain() {
        return new Booking(this.checkInDate, this.checkOutDate);
    }
}

File: src/main/kotlin/com/felipe/booking/alten/app/dto/BookingDTO.kt
package com.felipe.booking.alten.app.dto

import com.felipe.booking.alten.domain.model.Booking
import java.time.LocalDateTime

data class BookingDTO(
    val checkInDate: LocalDateTime,
    val checkOutDate: LocalDateTime,
) {

    companion object {
        fun of(booking: Booking) =
            BookingDTO(
                checkInDate =  booking.checkInDate,
                checkOutDate = booking.checkOutDate
            )
    }

    fun toDomain() =
        Booking(
            checkInDate = checkInDate,
            checkOutDate = checkOutDate
        )
}